OpenRA.Mods.D2k/Traits/SpiceBloom.cs

```diff
@@ -158,24 +153,22 @@ void SeedResources(Actor self)
 			}
 		}
 
-		public void Killed(Actor self, AttackInfo e)
+		void INotifyKilled.Killed(Actor self, AttackInfo e)
 		{
 			if (!string.IsNullOrEmpty(info.Weapon))
 				SeedResources(self);
 
-			self.World.AddFrameEndTask(t => t.Add(new DelayedAction(respawnTicks, () =>
+			var td = new TypeDictionary
 			{
-				var td = new TypeDictionary
-				{
-					new ParentActorInit(self),
-					new LocationInit(self.Location),
-					new CenterPositionInit(self.CenterPosition),
-					new OwnerInit(self.Owner),
-					new FactionInit(self.Owner.Faction.InternalName),
-					new SkipMakeAnimsInit()
-				};
-				self.World.CreateActor(info.SpawnActor, td);
-			})));
+				new ParentActorInit(self),
+				new LocationInit(self.Location),
+				new CenterPositionInit(self.CenterPosition),
+				new OwnerInit(self.Owner),
+				new FactionInit(self.Owner.Faction.InternalName),
+				new SkipMakeAnimsInit()
+			};
+
+			self.World.AddFrameEndTask(w => w.CreateActor(info.SpawnActor, td));
```

Comment:
Alright, maybe something for a future PR.  I feel that i've scope creeped this one far to much already.